tests/constraints/test_cardinality_constraint_continuous.py

```diff
@@ -12,37 +13,81 @@
 )
 from baybe.parameters.numerical import NumericalContinuousParameter
 from baybe.recommenders.pure.nonpredictive.sampling import RandomRecommender
-from baybe.searchspace.core import SearchSpace
+from baybe.searchspace.core import SearchSpace, SubspaceContinuous
 
 
-def _get_searchspace(
-    n_parameters: int, min_cardinality: int, max_cardinality: int
-) -> SearchSpace:
-    """Create a unit-cube searchspace with cardinality constraint on all parameters."""
-    parameters = [
+def _validate_samples(
+    samples: pd.DataFrame, max_cardinality: int, min_cardinality: int, batch_size: int
+):
+    """Validate whether samples fulfill certain conditions.
+
+    Conditions to check:
+    * cardinality condition
+    * have the right number of samples
+    * free of duplicates, except all zeros
+
+    Args:
+        samples: samples to check
+        max_cardinality: maximum cardinality
+        min_cardinality: minimum cardinality
+        batch_size: batch size of samples
+    """
+    # Assert that cardinality constraint is fulfilled
+    n_nonzero = np.sum(~np.isclose(samples, 0.0), axis=1)
+    assert np.all(n_nonzero >= min_cardinality) and np.all(n_nonzero <= max_cardinality)
+
+    # Assert that we obtain as many samples as requested
+    assert len(samples) == batch_size
+
+    # If there are duplicates, they must all come from the case cardinality = 0
+    assert np.all(samples[samples.duplicated()] == 0.0)
+
+
+# Combinations of cardinalities to be tested
+_cardinalities = sorted(combinations_with_replacement(range(0, 10), 2))
+
+
+@pytest.mark.parametrize(
+    "cardinalities", _cardinalities, ids=[str(x) for x in _cardinalities]
+)
+def test_sampling_cardinality_constraint(cardinalities):
+    """
```

Comment:
I'd prefer one of these three variants:
1. Full docstring
2. Short, one line docstring
3. No docstring but just a normal comment
All of these would remove the need of the `#noqa`

Reply:
The same applies to some other comments.

Reply:
Thanks for the options. I was not aware of option 3. 👍